edx_exams/apps/api/v1/tests/test_views.py

```diff
@@ -256,3 +256,101 @@ def test_exams_config(self, other_course_id, exam_type, expect_none_provider):
         self.assertEqual(new_exam.due_date, pytz.utc.localize(datetime.fromisoformat('2025-07-01 00:00:00')))
         self.assertEqual(new_exam.hide_after_due, False)
         self.assertEqual(new_exam.is_active, True)
+
+
+@ddt.ddt
+class CourseExamConfigurationsViewTests(ExamsAPITestCase):
+    """
+    Test CourseExamConfigurationsView
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        self.course_id = 'course-v1:edx+test+f19'
+
+        self.url = reverse('api:v1:exams-course_exam_configs', kwargs={'course_id': self.course_id})
+
+    # todo: can this helper method be moved to base?
+    def patch_api(self, user, data):
+        """
+        Helper function to make patch request to the API
+        """
+
+        data = json.dumps(data)
+        headers = self.build_jwt_headers(user)
+
+        return self.client.patch(self.url, data, **headers, content_type="application/json")
+
+    # todo: can this helper method be moved to base since both views use it?
+    def get_response(self, user, data, expected_response):
+        """
+        Helper functions to get API response
+        """
+        response = self.patch_api(user, data)
+        self.assertEqual(response.status_code, expected_response)
```

Comment:
Nit: This is feedback I gave on Marcin's pull request in `program-intent-engagement`, since it borrowed some functions from `edx-exams`.

Having an assertion in a function called `get_response` is unexpected to me. I think it would be more clear what a test is actually testing if the assertion is defined within the test. `self.get_response(self.user, 200)` doesn't really tell me the purpose of the `200`. It's not clear what the purpose of the `200` is in a method called `get_response`. `get_response`'s obvious responsibility is to make a request and return the response. If it also starts asserting things about the response, it's doing more than the name claims that it does, and it becomes harder to read, in my opinion.

An alternative would be to remove the line `self.assertEqual(response.status_code, expected_response)` and just make that assertion directly in your test. That way, it's clear in the actual test what you are testing. I think it would be clearer to collocate all assertions your test needs to make within the test.

Let me know what you think!

Reply:
Ah, this totally makes sense @MichaelRoytman ! I've updated to have the asserts be in the individual tests. Since the `get_response` method is now a little redundant, I've removed and just kept the `patch_api` helper method. Let me know if you think it should be renamed to be more clear though!